src/pages/Dashboard/scenes/Emails/Emails.tsx

```diff
@@ -64,19 +68,65 @@ const Emails: React.FC = () => {
   };
 
   const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    let selectedMentors: Mentor[] | undefined = [];
+    let selectedMentees: Mentee[] | undefined = [];
     switch (e.target.value) {
       case 'allMentors':
-        break;
-      case 'allMentees':
+        selectedMentors = mentors;
         break;
       case 'acceptedMentors':
+        selectedMentors = mentors?.filter(
+          (mentor) => mentor.state === 'approved'
+        );
+        break;
+      case 'pendingMentors':
+        selectedMentors = mentors?.filter(
+          (mentor) => mentor.state === 'pending'
+        );
+        break;
+      case 'rejectedMentors':
+        selectedMentors = mentors?.filter(
+          (mentor) => mentor.state === 'rejected'
+        );
+        break;
+      case 'allMentees':
+        selectedMentees = mentees?.filter(
+          (mentee) =>
+            mentee.state === 'approved' ||
+            mentee.state === 'pending' ||
+            mentee.state === 'rejected'
+        );
```

Comment:
here we don't need to filter right?  There are no other states where a mentee can be.